### ISSUE TITLE
Integrate posthog

### ISSUE THREAD

**Issue by PhilGarb** [open-legal-tech/open-decision#242]

**Comment by PhilGarb** [comment]
We want to integrate posthog to track user behavior in the open decision builder to better make product decisions.

**Comment by fbennets** [comment]
I think I'll build a custom transport for our centralized winston-based logger in the backend, that should be a straightforward implementation. 

https://posthog.com/docs/integrate/server/node

https://github.com/winstonjs/winston#adding-custom-transports

**Comment by PhilGarb** [comment]
So everything we log would be captured?

**Comment by fbennets** [comment]
Not necessarily, we can set a custom log level or specific flag to only log specific events to posthog.

**Comment by PhilGarb** [comment]
Ok. What would we log? It's pretty straight forward for the frontend, because it is autocapturing user interaction. But what would that look like on the server?

**Comment by fbennets** [comment]
For the beginning, I'd log all interactions with the auth and user APIs as well as mail sending.
In addition, logging the main steps in an app's lifecycle may make sense, e.g. creating, publishing and deleting an app.

**Comment by PhilGarb** [comment]
So we could tell how active a user is? We are logging most of this on the client already automatically when the user interacts. I am not sure how valuable an additional log from the server would be. Something like the emails send to a user is however very sensible. 
Maybe an approach like: what the user initiates is logged from the client and what we initiate is logged from the server.
